app/models/concerns/host_info_providers/static_info.rb

```diff
@@ -0,0 +1,68 @@
+module HostInfoProviders
+  module StaticInfo
+    extend ActiveSupport::Concern
+
+    included do
+      register_info_provider :static_info
+    end
+
+    def static_info
+      # Static parameters
+      param = {}
+      # maybe these should be moved to the common parameters, leaving them in for now
+      param["realm"] = realm.name unless realm.blank?
+      param["hostgroup"] = hostgroup.to_label unless hostgroup.blank?
+      param["comment"] = comment unless comment.blank?
+
+      add_network_params param
+      add_taxonomy_params param
+      add_domain_params param
+      add_login_params param
+      add_unattended_params param
+
+      # Parse ERB values contained in the parameters
+      param = SafeRender.new(:variables => { :host => self }).parse(param)
+
+      { 'parameters' => param }
+    end
+
+    private
+
+    def add_unattended_params(param)
+      return unless SETTINGS[:unattended]
+
+      param["root_pw"] = root_pass unless (!operatingsystem.blank? && operatingsystem.password_hash == 'Base64')
+    end
+
+    def add_login_params(param)
+      return unless SETTINGS[:login] && owner
+
+      param["owner_name"]  = owner.name
+      param["owner_email"] = owner.is_a?(User) ? owner.mail : owner.users.map(&:mail)
+    end
+
+    def add_domain_params(param)
+      return if domain.blank?
+      param["domainname"] = domain.name unless domain.name.blank?
+      param["foreman_domain_description"] = domain.fullname unless domain.fullname.blank?
+    end
+
+    def add_taxonomy_params(param)
+      if SETTINGS[:locations_enabled]
+        param["location"] = location.title unless location.blank?
```

Comment:
this now needs to look a little different, please see b96e021026cdcc6918c61d246dbf5c7baabfc1b4